### PR TITLE
upgrade to iojs 4.0.0

### PR DESCRIPTION
**Pull request by calvinmetcalf**

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
so it's really unclear why the iDevices started fail so consistently...

**Comment by calvinmetcalf** [comment]
since I was still trying to get iDevices working figured I'd up it to 3.0.0

**Comment by calvinmetcalf** [comment]
updated to 4.0.0 rc.1, this actually has some things changed besides formatting (with regards to listenerCount and high water mark) so there is now some actual incentive for me to get this published.  I believe the current issues have more to do with sauce labs then the library but I will investigate more.

**Comment by calvinmetcalf** [comment]
and we are finally green!

**Comment by calvinmetcalf** [comment]
ok @nodejs/streams this is ready for review

**Comment by calvinmetcalf** [comment]
@chrisdickinson, @domenic, @dcousens anyone?

File: doc/stream.markdown [hunk 1]
```diff
@@ -1442,7 +1487,7 @@ code) will know when to check again, by calling `stream.read(0)`.  In
 those cases, you *may* call `stream.push('')`.
 
 So far, the only use case for this functionality is in the
-[tls.CryptoStream][] class, which is deprecated in io.js v1.0.  If you
+[tls.CryptoStream][] class, which is deprecated in Node.js/io.js v1.0.  If you
```
Comment (dcousens) [inline doc/stream.markdown]: Node.js `v1.0`?

**Comment by dcousens** [comment]
LGTM, +1

File: doc/stream.markdown (hunk 1 above)
Comment (calvinmetcalf) [inline doc/stream.markdown]: this is taken from the node docs, but io.js 1.0 was retroactively made node.js 1.0

**Pull request merged**